src/main/java/bhs/devilbotz/subsystems/DriveTrain.java

```diff
@@ -470,4 +527,54 @@ public double getRoll() {
   public double getYaw() {
     return navx.getYaw();
   }
+
+  /**
+   * Uses ramsete controller to follow the specified trajectory
+   *
+   * @param traj Requested trajectory
+   * @param isFirstPath Set to true if this is the first path being run in autonomous in order to
+   *     reset odometry before starting
+   * @return A sequential command that when executed, moves the robot along the specified trajectory
+   * @see <a
+   *     href=https://github.com/mjansen4857/pathplanner/wiki/PathPlannerLib:-Java-Usage#ppramsetecommand>PathPlanner
+   *     Example</a>
+   */
+  public Command followTrajectoryCommand(PathPlannerTrajectory traj, boolean isFirstPath) {
```

Comment:
If this is permanent code, maybe convert this to a command file instead of having an inline command declaration

Reply:
I'm OK either way.  The nice thing about having the command generation inside of the DriveTrain is that the details of the feedforward, PID values, etc don't need to be exposed to the upper layer.  So the autonomous commands could call this multiple times to chain different paths together in a sequence.  This is also copied directly from the PathPlanner example.  What's the benefit of having this in a command file?

Reply:
See https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html
Might help compare/contrast the different methods.  I also like how everything is in one file. I tried moving this kind of code into separate file and found it hard/tedious because of all the things that had to be exposed.
I think for this right now leaving it as is in subsystem is fine.  Once we figure out how/what we will do with path planning we could move this out of here or adjust what method returns?

Reply:
Looking at the autonomous commands that the strategy team sent out, I feel that the "commands" directory should contain combinations of commands from the other subsystems (drivetrain, arm, and gripper).  E.g. the gripper subsystem has methods to open/close the gripper.  Similarly, the arm will have commands to raise/lower the gripper to a specified angle (and rate)?  So, I see "followTrajectoryCommand" similar in the sense that it doesn't define specifics, but just a method that moves the robot in a particular path that could be used by the other mechanisms.   So, the "Routine 3: Score and Mobility" autonomous command might be structured as follows:

In parallel:
 - move to scoring zone (_using drivetrain.followTrajectory()_)
 - raise arm to middle scoring position (_using Arm.moveToAngle?_)
When both commands are done:
- open gripper (_using Gripper.open_)